rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java

```diff
@@ -924,6 +926,93 @@ void saveFederationElection_shouldSaveInStorage() {
         assertArrayEquals(federationElectionSerializedWithVote, actualAbiCallElectionSerialized);
     }
 
+    @Test
+    void saveFundTransactionUnsignedHash_preLovell700_shouldNotSaveInStorage() {
+
+        // Arrange
+
+        ActivationConfig.ForBlock arrowheadActivations = ActivationConfigsForTest.arrowhead600().forBlock(0L);
+
+        StorageAccessor storageAccessor = new InMemoryStorage();
+        FederationStorageProvider federationStorageProvider = new FederationStorageProviderImpl(storageAccessor);
+
+        Sha256Hash expectedHash = BitcoinTestUtils.createHash(123_456_789);
+
+        federationStorageProvider.setFundTransactionUnsignedHash(expectedHash);
+
+        // Act
+
+        federationStorageProvider.save(networkParameters, arrowheadActivations);
+
+        // Assert
+
+        byte[] actualHashBytes = storageAccessor.getFromRepository(FUND_TX_HASH_UNSIGNED.getKey(), data -> data);
+
+        assertNull(actualHashBytes);
+
+    }
+
+    @Test
+    void saveFundTransactionUnsignedHash_postLovell700_shouldSaveInStorage() {
+
+        // Arrange
+
+        ActivationConfig.ForBlock lovell700Activations = ActivationConfigsForTest.lovell700().forBlock(0L);
```

Comment:
Should we use lovell or `all`? Also this is probably gonna be used a lot, maybe it should be part of the class. Or this whole set of tests could be Nested

Reply:
Updated.